api/cmd/portainer/main.go

```diff
@@ -100,6 +101,14 @@ func retrieveFirstEndpointFromDatabase(endpointService portainer.EndpointService
 	return &endpoints[0]
 }
 
+func retrievePassword(flags *portainer.CLIFlags) string {
+  password := *flags.AdminPassword
+  if !strings.HasPrefix(password, "admin:") {
```

Comment:
I thought about putting it in `ValidateFlags` for consistency but stripping the `admin:` preface in that method without checking its presence feels weird. If I saw the sanitation without the validation in that method I would assume it's a bug.

If tomorrow the user for the password changed from `admin:` to something else it wouldn't be obvious to developers that they need to change the code in two quite different code paths.